app/controllers/concerns/fully_authenticatable.rb

```diff
@@ -0,0 +1,14 @@
+module FullyAuthenticatable
+  def confirm_two_factor_authenticated(id = nil)
+    return redirect_to sign_up_start_url(request_id: id) unless user_signed_in?
+
+    return prompt_to_set_up_2fa unless current_user.two_factor_enabled?
+
+    prompt_to_enter_otp
```

Comment:
Here you go: https://github.com/18F/identity-idp/pull/1265/commits/d1ba3b9ee598cd570b06377973e176391674666e
Scroll down to loa1_sso_spec